### PR TITLE
javadocs for utils

### PR DESCRIPTION
**Pull request by shilparaaju**

### CODE REVIEW

File: src/main/java/in/ac/bits/protocolanalyzer/utils/Security.java [hunk 1]
```diff
@@ -4,9 +4,27 @@
 import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
 
+/**
+ * This utility class is used to provide security.
```
Comment (prasadtalasila) [inline src/main/java/in/ac/bits/protocolanalyzer/utils/Security.java]: The class contains wrapper functions around cryptographic algorithms.

File: src/main/java/in/ac/bits/protocolanalyzer/utils/Security.java [hunk 2]
```diff
@@ -4,9 +4,27 @@
 import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
 
+/**
+ * This utility class is used to provide security. 
+ * For this purpose it uses the SHA-256 hash function.
+ *
+ * @author Shilpa Raju
```
Comment (prasadtalasila) [inline src/main/java/in/ac/bits/protocolanalyzer/utils/Security.java]: Please add the name of __crygnus (Mihir Kakrambe)__ as one of the authors to all the files.

File: src/main/java/in/ac/bits/protocolanalyzer/utils/ByteOperator.java [hunk 3]
```diff
@@ -1,7 +1,22 @@
 package in.ac.bits.protocolanalyzer.utils;
 
+/**
+ * This utility allows for various byte operations.
```
Comment (prasadtalasila) [inline src/main/java/in/ac/bits/protocolanalyzer/utils/ByteOperator.java]: The class contains utility methods for converting byte range into chosen primitive data types (int, short, long etc.).

File: src/main/java/in/ac/bits/protocolanalyzer/utils/BitOperator.java [hunk 4]
```diff
@@ -30,11 +31,10 @@ public static int getBit(byte b, int index)
     }
 
     /**
-     * Returns int array 'bit' of size 8 with bit[i] = bit at index 'i' in byte
-     * b
+     * Converts the given byte to an integer array.
```
Comment (prasadtalasila) [inline src/main/java/in/ac/bits/protocolanalyzer/utils/BitOperator.java]: please provide an example here.

File: src/main/java/in/ac/bits/protocolanalyzer/utils/BitOperator.java [hunk 5]
```diff
@@ -48,11 +48,11 @@ public static int getBit(byte b, int index)
      * Returns int equivalent of bit array of specified start location and
      * length
      * 
-     * @param b
-     * @param startBit
-     * @param numberOfBits
-     * @return int val
-     * @throws ArrayIndexOutOfBoundsException
+     * @param b This is the byte from which values are obtained
```
Comment (prasadtalasila) [inline src/main/java/in/ac/bits/protocolanalyzer/utils/BitOperator.java:54]: a summary description of the method is missing here.

File: src/main/java/in/ac/bits/protocolanalyzer/utils/BitOperator.java [hunk 6]
```diff
@@ -73,10 +73,10 @@ public static int getValue(byte b, int startBit, int numberOfBits)
      * Returns integer equivalent of the nibble in byte b with given nibble
      * index (either 0 or 1)
      * 
-     * @param b
-     * @param nibbleIndex
-     * @return int nibble
-     * @throws ArrayIndexOutOfBoundsException
```
Comment (prasadtalasila) [inline src/main/java/in/ac/bits/protocolanalyzer/utils/BitOperator.java:79]: a summary description of the method is missing here.

File: src/main/java/in/ac/bits/protocolanalyzer/utils/BitOperator.java [hunk 7]
```diff
@@ -89,6 +89,11 @@ public static int getNibble(byte b, int nibbleIndex)
                     "Nibble index can only be 0 or 1");
         }
     }
+
```
Comment (prasadtalasila) [inline src/main/java/in/ac/bits/protocolanalyzer/utils/BitOperator.java:96]: This method takes a byte array and produces a smaller byte array indicated by the arguments. If the operation of the method is not clear, please write a skeletal example using the same class. That should make the operation of this method very obvious.

Comment (prasadtalasila) [reply src/main/java/in/ac/bits/protocolanalyzer/utils/BitOperator.java:96]: Please follow this advice of constructing minimum working example (MWE) for all the methods that are not clear.

**Review by prasadtalasila — CHANGES_REQUESTED** [review]
@shilparaaju 
Please don't add the javadoc to the code base. You can write a shell script that would generate the documentation and stores the same in docs/ folder. The script can be named as __gen_doc.sh__ and placed in __scripts/__ directory.

Right now, the script is not considering the package structure of in.ac.bits.protocolanalyzer.util package. You can use the following command to create javadoc with proper package structure.
```bash
$javadoc -version -author -private -d docs -classpath "<lib-path>/*" -sourcepath . -subpackages .
```
You should also create a package-info.java for util package.

**Comment by shilparaaju** [comment]
@prasadtalasila I have made the changes requested. Please let me know if there is anything else I have to change.

**Comment by prasadtalasila** [comment]
@shilparaaju The following points still need improvement.

1. As per the pending comments, a few methods still have their description missing.
1. The package-info.java requires some introduction to overall utility of the package.
1. The Javadoc html files need to be removed from the code base.

**Comment by shilparaaju** [comment]
@prasadtalasila 
I have made the changes. Please let me know if I missed out on anything.

File: scripts/gen_doc.sh [hunk 8]
```diff
@@ -0,0 +1,2 @@
+javadoc -version -author -private -d docs -classpath /home/ubuntu/darshini/src/main/java/in/ac/bits/protocolanalyzer/utils/* -sourcepath . -subpackages .
```
Comment (prasadtalasila) [inline scripts/gen_doc.sh]: it is better to specify this as a relative path to the location of the script.

**Review by prasadtalasila — CHANGES_REQUESTED** [review]
@shilparaaju please make the suggested changes.

@raghavagrawal please read through the javadoc put in this commit and make comments if any of it is not clear to you. If everything is okay, then you can comment with okay status.

**Comment by shilparaaju** [comment]
@prasadtalasila I made the change requested

**Pull request merged**